Move `typescript` package to peerDependencies?

This proposes to move `typescript` package from `dependencies` to `peerDependencies` in `package.json`.

Because `typescript` multiple versions can be installed in some environments:

```
$ cat package.json
{
  "dependencies": {
    "typescript": "^3.1.6",
    "tyscan": "^0.1.1"
  }
}

$ npm ls typescript
sideci@ /Users/foo/git/sample
├── typescript@3.1.6
└─┬ tyscan@0.1.1
  └── typescript@3.2.4  deduped
```

https://github.com/sider/TyScan/blob/91672f19239e5575174ee865f302a947e7c224e7/package.json#L32

For example:
```json
{
  "dependencies": {
    "typescript": "^3.2.4"
  }
}
```
⬇️ 
```json
{
  "devDependencies": {
    "typescript": "^3.2.4"
  },
  "peerDependencies": {
    "typescript": "2 || 3"
  }
}
```

See also:
- https://docs.npmjs.com/files/package.json#peerdependencies
- https://nodejs.org/en/blog/npm/peer-dependencies/

Thanks for the quick response! I'm trying your suggested command `npm install 'github:sider/TyScan#fix/issue2'`, but I encounter the error below:

```
$ npm install 'github:sider/TyScan#fix/issue2'
npm ERR! prepareGitDep 1>
npm ERR! prepareGitDep > tyscan@0.1.1 prepare /Users/koba/.npm/_cacache/tmp/git-clone-b3bc8d35
npm ERR! prepareGitDep > webpack --config webpack.config.js
npm ERR! prepareGitDep
npm ERR! prepareGitDep
npm ERR! prepareGitDep 2> npm WARN install Usage of the `--dev` option is deprecated. Use `--only=dev` instead.
npm ERR! prepareGitDep sh: webpack: command not found
npm ERR! prepareGitDep npm ERR! file sh
npm ERR! prepareGitDep npm ERR! code ELIFECYCLE
npm ERR! prepareGitDep npm ERR! errno ENOENT
npm ERR! prepareGitDep npm ERR! syscall spawn
npm ERR! prepareGitDep npm ERR! tyscan@0.1.1 prepare: `webpack --config webpack.config.js`
npm ERR! prepareGitDep npm ERR! spawn ENOENT
npm ERR! prepareGitDep npm ERR!
npm ERR! prepareGitDep npm ERR! Failed at the tyscan@0.1.1 prepare script.
npm ERR! prepareGitDep npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! prepareGitDep
npm ERR! prepareGitDep npm ERR! A complete log of this run can be found in:
npm ERR! prepareGitDep npm ERR!     /Users/koba/.npm/_logs/2019-01-30T13_43_36_766Z-debug.log
npm ERR! prepareGitDep
npm ERR! premature close

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/koba/.npm/_logs/2019-01-30T13_43_36_907Z-debug.log
```

The error log's content is:

```
$ cat /Users/koba/.npm/_logs/2019-01-30T13_43_36_907Z-debug.log
0 info it worked if it ends with ok
1 verbose cli [ '/usr/local/Cellar/node/11.8.0/bin/node',
1 verbose cli   '/usr/local/bin/npm',
1 verbose cli   'install',
1 verbose cli   'github:sider/TyScan#fix/issue2' ]
2 info using npm@6.7.0
3 info using node@v11.8.0
4 verbose npm-session e9682de4c84c4049
5 silly install loadCurrentTree
6 silly install readLocalPackageData
7 verbose prepareGitDep github:sider/TyScan#fix/issue2: installing devDeps and running prepare script.
8 error prepareGitDep 1>
8 error prepareGitDep > tyscan@0.1.1 prepare /Users/koba/.npm/_cacache/tmp/git-clone-b3bc8d35
8 error prepareGitDep > webpack --config webpack.config.js
9 error prepareGitDep 2> npm WARN install Usage of the `--dev` option is deprecated. Use `--only=dev` instead.
9 error prepareGitDep sh: webpack: command not found
9 error prepareGitDep npm ERR! file sh
9 error prepareGitDep npm ERR! code ELIFECYCLE
9 error prepareGitDep npm ERR! errno ENOENT
9 error prepareGitDep npm ERR! syscall spawn
9 error prepareGitDep npm ERR! tyscan@0.1.1 prepare: `webpack --config webpack.config.js`
9 error prepareGitDep npm ERR! spawn ENOENT
9 error prepareGitDep npm ERR!
9 error prepareGitDep npm ERR! Failed at the tyscan@0.1.1 prepare script.
9 error prepareGitDep npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
9 error prepareGitDep
9 error prepareGitDep npm ERR! A complete log of this run can be found in:
9 error prepareGitDep npm ERR!     /Users/koba/.npm/_logs/2019-01-30T13_43_36_766Z-debug.log
10 silly fetchPackageMetaData error for github:sider/TyScan#fix/issue2 premature close
11 timing stage:rollbackFailedOptional Completed in 1ms
12 timing stage:runTopLevelLifecycles Completed in 8578ms
13 verbose stack Error: premature close
13 verbose stack     at PassThrough.onclose (/usr/local/lib/node_modules/npm/node_modules/end-of-stream/index.js:47:67)
13 verbose stack     at PassThrough.emit (events.js:202:15)
13 verbose stack     at emitCloseNT (internal/streams/destroy.js:59:8)
13 verbose stack     at processTicksAndRejections (internal/process/next_tick.js:76:17)
14 verbose cwd /Users/koba/tmp/hoge
15 verbose Darwin 18.2.0
16 verbose argv "/usr/local/Cellar/node/11.8.0/bin/node" "/usr/local/bin/npm" "install" "github:sider/TyScan#fix/issue2"
17 verbose node v11.8.0
18 verbose npm  v6.7.0
19 error premature close
20 verbose exit [ 1, true ]
```

--- a/package.json
+++ b/package.json
@@ -27,9 +27,7 @@
     "fast-glob": "^2.2.6",
     "js-yaml": "^3.12.1",
     "parsimmon": "^1.12.0",
-    "tsconfig": "^7.0.0",
-    "tsutils": "^3.7.0",
-    "typescript": "^3.2.4"
+    "tsconfig": "^7.0.0"
   },
   "devDependencies": {
     "@types/chai": "^4.1.7",
@@ -46,7 +44,12 @@
     "ts-node": "^8.0.1",
     "tslint": "^5.12.1",
     "tslint-config-airbnb": "^5.11.1",
+    "tsutils": "^3.7.0",
+    "typescript": "^3.2.4",
     "webpack-merge": "^4.2.1"
+  },
+  "peerDependencies": {
+    "typescript": "2 || 3"
   },
   "nyc": {
     "include": [

